core/model/configuration.py

```diff
@@ -1132,6 +1132,15 @@ def __get__(
 
         if setting_value is None:
             setting_value = self.default
+        elif self.type == ConfigurationAttributeType.NUMBER:
+            # In case we're a number, the default should replace any incorrect formatting
+            try:
+                setting_value = float(setting_value)
+            except ValueError:
+                logging.getLogger(self.__class__.__name__).error(
+                    f"Could not typecast {self.label} value '{setting_value}'. Falling back to default"
+                )
+                setting_value = self.default
```

Comment:
Also, are we sure that `self.default` is a value that is a valid `…NUMBER`? What happens if it has been set to something invalid?

Reply:
For now the breadth doesn't quite exist, since the UI will only allow empty, None or a number value. This is only possible if the API is hit directly.